### PR TITLE
Fixed 404 error when connecting to VSO

### PR DESCRIPTION
**Pull request by rdagumampan**
Substituted the project collection name with DefaultCollection as defined in VSO REST API documentation.
Without this, we received 404 error because the path is non existent.

Reference here:
https://www.visualstudio.com/en-us/docs/integrate/api/tfs/projects

### CODE REVIEW

**Comment by lprichar** [comment]
Hi @rdagumampan, thank you so much for your contribution! And I'm sorry it took so long to get back to you.  It's very hard to tell what changed in this diff.  Did you by chance change the line endings?  Would it be inconvenient for you to change them back?

**Comment by rdagumampan** [comment]
i realized that too, seems the diffgram didn't work well in this. Basicallt, i introduced a method at the end
SubstituteName(...) which replaces the project collection name into "Default" when it sees its the same name as the VSO sub domain. 

Let me try again later, maybe R# reformatted the codes.

File: TfsRestServices/TfsRestService.cs [hunk 1]
```diff
@@ -1,97 +1,118 @@
...
+            var message = firstComment?.Message;
+            return message;
+        }
+
+        private string SubstituteName(Uri url, string projectCollectionName)
+        {
+            if (url.HostNameType == UriHostNameType.Dns)
+            {
+                var projectNameFromUrl = url.Host.Split('.')[0];
+                return string.Equals(projectNameFromUrl, projectCollectionName) ? "DefaultCollection" : projectNameFromUrl;
```
Comment (lprichar) [inline TfsRestServices/TfsRestService.cs]: Just about to merge this.  Adding a unit test, but something doesn't look right.  Shouldn't this line be `return string.Equals(projectNameFromUrl, projectCollectionName) ? "DefaultCollection" : projectCollectionName;`?

Comment (lprichar) [inline TfsRestServices/TfsRestService.cs]: Actually I'm going to merge because I don't want to delay this PR any longer (sorry it took so long, I've been super busy at work).  But if you get a sec can you please look over this: https://github.com/AutomatedArchitecture/SirenOfShame/commit/1da528978daa99f437ef44113b9bf30a2a30d80b

Comment (rdagumampan) [inline TfsRestServices/TfsRestService.cs]: @lprichar , yes, this is mess completely forgot to cover with tests. You're right it should return projectCollectionName. Thanks.
